lib/curl_sha512_256.c

```diff
@@ -52,6 +52,22 @@
 #      include <openssl/evp.h>
 #      define USE_OPENSSL_SHA512_256          1
 #      define HAS_SHA512_256_IMPLEMENTATION   1
+#      ifdef __NetBSD__
+#        include <sys/param.h>
+#        if __NetBSD_Version__ <=  903000000 || \
+           (__NetBSD_Version__ >=  999000000 && \
+            __NetBSD_Version__ <  1000000000) ||
+           (__NetBSD_Version__ >= 1099000000 && \
+            __NetBSD_Version__ <= 1099001000)
```

Comment:
Right. That's why `__NetBSD_Version__ <=  903000000` means `has bug` :)
Note **`<=`**

Reply:
Thanks for the information.
I'll update the version checks accordingly.

Reply:
These macros are used only if supported version of OpenSSL is detected.
It also safe to apply the workaround even if it is not required.
So the simplified checks:
* Any versions before 9.4 have the bug
* Versions from 9.99 until 10.0 (excluding) have the bug
* Versions from 10.99.0 until 10.99.11 (excluding) have the bug
* Otherwise these is no bug

Dev versions could be used for various testing. It is easy to support them.

Reply:
>    Any versions before 9.4 have the bug
    Versions from 9.99 until 10.0 (excluding) have the bug
    Versions from 10.99.0 until 10.99.11 (excluding) have the bug
    Otherwise these is no bug

Maybe you can add that textual version as a comment? This huge preprocessor conditions list is quite a difficult thing to parse in your head without some guidance!

Reply:
@bagder Comments have been added.